xtask/src/changelog.rs

```diff
@@ -0,0 +1,369 @@
+use std::fs;
+use std::path::Path;
+
+use crate::util::{project_dir, write_to_string};
+use clap::Args;
+use cross::shell::MessageInfo;
+use cross::ToUtf8;
+use eyre::Context;
+use serde::Deserialize;
+
+#[derive(Args, Debug)]
+pub struct BuildChangelog {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Do not print cross log messages.
+    #[clap(short, long)]
+    pub quiet: bool,
+    /// Whether messages should use color output.
+    #[clap(long)]
+    pub color: Option<String>,
+    /// Build a draft changelog (to `CHANGELOG.md.draft`).
+    #[clap(short, long)]
+    draft: bool,
+}
```

Comment:
this commands need to support `cargo release <v>`, which is a dry-run.

It should also take into account if the release is a pre-release, and can also take the new version and previous version to see if it's a breaking change, and error if there is a breaking change but not a breaking bump.


`cargo-release` defines a [few environment variables](https://github.com/crate-ci/cargo-release/blob/master/docs/reference.md#hook-environment-variables) that can help